src/async_wrap.cc

```diff
@@ -857,15 +852,6 @@ void AsyncWrap::AsyncReset(Local<Object> resource, double execution_async_id,
                                                      : execution_async_id;
   trigger_async_id_ = env()->get_default_trigger_async_id();
 
-  {
-    HandleScope handle_scope(env()->isolate());
-    Local<Object> obj = object();
-    CHECK(!obj.IsEmpty());
-    if (resource != obj) {
-      USE(obj->Set(env()->context(), env()->resource_symbol(), resource));
-    }
-  }
```

Comment:
@Qard Thank you for the explanation. Does the test failure in https://github.com/nodejs/node/pull/38468/commits/df97c7e5b1dc7bba033758629dfbf19857729d72:
```console
npm ERR! stream[kUpdateTimer] is not a function
```
occur because of the handle reuse you mentioned?
https://github.com/nodejs/node/blob/4ebb88fea5a9d83f2c4ec566e958d735f1fca138/lib/_http_agent.js#L528-L529
The assigned object does not have a `[kUpdateTimer]` key which I guess the reused socket handle is expected to have.

Since you mentioned the differences, I'm wondering if we should merge them as they seem to serve different purposes. :eyes: